### PR TITLE
Save ram

### PR DESCRIPTION
**Pull request by sgliner-ledger**
This optimization brings back the swap feature on LNS

### CODE REVIEW

File: src/btchip_apdu_get_coin_version.c [hunk 1]
```diff
@@ -31,19 +31,19 @@ unsigned short btchip_apdu_get_coin_version() {
...
-    G_io_apdu_buffer[offset++] = strlen(G_coin_config->name_short);
-    os_memmove(G_io_apdu_buffer + offset, G_coin_config->name_short,
-               strlen(G_coin_config->name_short));
-    offset += strlen(G_coin_config->name_short);
+    G_io_apdu_buffer[offset++] = COIN_P2PKH_VERSION >> 8;
+    G_io_apdu_buffer[offset++] = COIN_P2PKH_VERSION;
+    G_io_apdu_buffer[offset++] = COIN_P2SH_VERSION >> 8;
+    G_io_apdu_buffer[offset++] = COIN_P2SH_VERSION;
+    G_io_apdu_buffer[offset++] = COIN_FAMILY;
+    G_io_apdu_buffer[offset++] = strlen(COIN_COINID);
```
Comment (xchapron-ledger) [inline src/btchip_apdu_get_coin_version.c]: You can replace all `strlen(<static string>)` by `sizeof(<static string>) - 1`. This spare a very little bit of computing and flash.

File: src/btchip_apdu_hash_sign.c [hunk 2]
```diff
@@ -183,14 +183,14 @@ void btchip_bagl_user_action_signtx(unsigned char confirming, unsigned char dire
...
             cx_hash(&btchip_context_D.transactionHashFull.sha256.header, CX_LAST,
                 hash, 0, hash, 32);
             PRINTF("Hash1\n%.*H\n", sizeof(hash), hash);
 
             // Rehash
-            cx_sha256_init(&localHash);
-            cx_hash(&localHash.header, CX_LAST, hash, sizeof(hash), hash, 32);
+            // Use a global context already existing to save some ram
+            cx_sha256_init(&btchip_context_D.hashAmount.sha256);
+            cx_hash(&btchip_context_D.hashAmount.sha256.header, CX_LAST, hash, sizeof(hash), hash, 32);
```
Comment (xchapron-ledger) [inline src/btchip_apdu_hash_sign.c:182]: Why not reusing `&btchip_context_D.transactionHashFull.sha256.header` or even better, using directly `cx_hash_sha256` for both hash iteration as they are one shot hash.

File: src/btchip_apdu_get_coin_version.c (hunk 1 above)
Comment (xchapron-ledger) [reply src/btchip_apdu_get_coin_version.c]: They are other places in this commit where `strlen` can be replaced.

**Review by xchapron-ledger — APPROVED** [review]
Minors nitpicking, but looks good to me.

File: src/btchip_apdu_hash_sign.c (hunk 2 above)
Comment (sgliner-ledger) [reply src/btchip_apdu_hash_sign.c:182]: TBH I don't know why it was done using a local buffer before, but I cannot test the code that easily :upside_down_face:

**Pull request merged**